api/envoy/config/filter/http/router/v2/router.proto

```diff
@@ -26,4 +26,10 @@ message Router {
   // an upstream request. Presuming retries are configured, multiple upstream
   // requests may be made for each downstream (inbound) request.
   repeated envoy.config.filter.accesslog.v2.AccessLog upstream_log = 3;
+
+  // Do not add any additional *x-envoy-* headers to requests or responses. This
+  // only affects the router filter generated *x-envoy-* headers, other Envoy
+  // filters and the HTTP connection manager may continue to set *x-envoy-*
+  // headers.
+  bool suppress_envoy_headers = 4;
```

Comment:
I think my comment was buried here since you updated it. Here it is again:

Just for posterity, a different solution would be to add this at the HTTP connection manager level, and expose it as a filter callback setting. We discussed this in some other comment thread. I don't have a strong opinion one way or the other. Can you do a doc link to the router docs where we describe which headers are set? https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/router_filter#http-headers Bonus points for making it more clear somehow which are set and which are read.

Reply:
I agree this could be callback, but I think it would fundamentally break many filters, since they rely on `x-envoy-*` headers for their behavior. I opted for a light touch "do the minimum necessary to build a solution where we can sanitize" solution here.